File: frontend/src/auth/roles.ts
// Mirrors the backend Roles class. Add new roles here when the backend gains them.
export const ROLES = {
  Administrator: 'Administrator',
  WarehouseManager: 'WarehouseManager',
  ProcurementOfficer: 'ProcurementOfficer',
  Employee: 'Employee',
} as const;

// Human-friendly role labels now live in the i18n dictionary (role.* keys),
// so components translate them with t(`role.${role}`) instead of a static map.

// Roles allowed to create/edit/delete data.
export const MANAGER_ROLES: string[] = [ROLES.Administrator, ROLES.WarehouseManager];

// Roles that handle buying and stock: suppliers, purchases, stock movements,
// material plans and store items (the managers + the Procurement Officer).
export const PROCUREMENT_ROLES: string[] = [...MANAGER_ROLES, ROLES.ProcurementOfficer];

export const canManage = (role?: string) => !!role && MANAGER_ROLES.includes(role);
export const canProcure = (role?: string) => !!role && PROCUREMENT_ROLES.includes(role);
export const isAdmin = (role?: string) => role === ROLES.Administrator;

File: backend/CompanySystem.Api/Auth/Roles.cs
namespace CompanySystem.Api.Auth;

/// <summary>
/// All roles live here in ONE place. To add a new role later:
///   1) add a const below,
///   2) add it to All,
///   3) (optionally) include it in Managers if it may modify data.
/// Everything else (login, [Authorize(Roles=...)], the Users page) reads from here.
/// </summary>
public static class Roles
{
    public const string Administrator = "Administrator";
    public const string WarehouseManager = "WarehouseManager";
    public const string ProcurementOfficer = "ProcurementOfficer";
    public const string Employee = "Employee";

    // Every role the system knows about.
    public static readonly string[] All = { Administrator, WarehouseManager, ProcurementOfficer, Employee };

    // Roles allowed to create/update/delete data. Extend this as new roles gain write access.
    public const string Managers = Administrator + "," + WarehouseManager;

    // Roles that handle buying and stock: suppliers, purchases, stock movements,
    // material plans and store items. The managers, plus the Procurement Officer
    // (who can't touch employees, projects or the cost reports).
    public const string Procurement = Managers + "," + ProcurementOfficer;

    // Human-friendly labels (the raw value has no spaces to keep tokens clean).
    public static string DisplayName(string role) => role switch
    {
        WarehouseManager => "Warehouse Manager",
        ProcurementOfficer => "Procurement Officer",
        _ => role,
    };

    public static bool IsValid(string role) => System.Array.IndexOf(All, role) >= 0;
}
